hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/MiniOzoneClusterImpl.java

```diff
@@ -138,23 +138,32 @@ public class MiniOzoneClusterImpl implements MiniOzoneCluster {
   private final Set<AutoCloseable> clients = ConcurrentHashMap.newKeySet();
   private SecretKeyClient secretKeyClient;
 
+  // TODO: Get rid of Optional, just use null
+  private Optional<int[]> dnInitialVersion = Optional.empty();
+  private Optional<int[]> dnCurrentVersion = Optional.empty();
+
   /**
    * Creates a new MiniOzoneCluster with Recon.
    *
    * @throws IOException if there is an I/O error
    */
+  @SuppressWarnings("checkstyle:ParameterNumber")
   MiniOzoneClusterImpl(OzoneConfiguration conf,
                        SCMConfigurator scmConfigurator,
                        OzoneManager ozoneManager,
                        StorageContainerManager scm,
                        List<HddsDatanodeService> hddsDatanodes,
-                       ReconServer reconServer) {
+                       ReconServer reconServer,
+                       Optional<int[]> dnInitialVersion,
+                       Optional<int[]> dnCurrentVersion) {
```

Comment:
The two arrays passed here is not at the same level of abstraction as the other parameters.  Handling initial and current versions would be better done when creating the `HddsDatanodeService` instances.  That way we could avoid cluttering the `MiniOzoneCluster` interface.